lib/sign_in/widgets/form.dart

```diff
@@ -82,48 +67,62 @@ class _SignInFormState extends State<SignInForm> {
               alignment: Alignment.bottomCenter,
             ),
           ),
-          SizedBox(height: 16),
-          SignInInput(
-            controller: _emailController,
-            label: s.signIn_form_email,
-            error: _isEmailValid ? null : s.signIn_form_email_error,
-            onChanged: () => setState(() {}),
-          ),
-          SizedBox(height: 16),
-          SignInInput(
-            controller: _passwordController,
-            label: s.signIn_form_password,
-            obscureText: true,
-            error: _isPasswordValid ? null : s.signIn_form_password_error,
-            onChanged: () => setState(() {}),
-          ),
-          SizedBox(height: 16),
-          MorphingLoadingButton(
-            onPressed: _signIn,
-            isLoading: _isLoading,
-            child: Padding(
-              padding: EdgeInsets.all(12),
-              child: Text(
-                _isLoading ? s.general_loading : s.signIn_form_signIn,
-                style: TextStyle(color: Colors.white, fontSize: 20),
-              ),
+          SizedBox(height: 32),
+          ConstrainedBox(
+            constraints: BoxConstraints(
+              maxHeight: mediaQuery.size.height -
+                  400 -
+                  mediaQuery.padding.bottom -
+                  mediaQuery.padding.top,
             ),
+            child: WebView(
+                javascriptMode: JavascriptMode.unrestricted,
+                onWebViewCreated: (controller) => this.controller = controller,
+                onPageFinished: (url) async {
+                  if (url ==
+                      'https://${services.get<AppConfig>().domain}/login') {
+                    // The JavaScript is meant to isolate the login section
+                    // Hopefully there will be an option to only get that
+                    // in a non-hacky way in the future.
+                    await controller.evaluateJavascript('''
+                  var node = document.getElementById('loginarea');
+                  var html = document.getElementsByTagName('html')[0];
+                  html.removeChild(html.childNodes[2]);
+                  html.appendChild(node);
+                  document.getElementsByTagName('h2')[0].innerHTML = ''
+                  ''');
+                  } else if (url ==
+                      'https://${services.get<AppConfig>().domain}/dashboard') {
```

Comment:
Can we already intercept this in `WebView.navigationDelegate`? Currently, we see a flash of the web-dashboard and are then forwarded to the app-dashboard.